Headlines/src/Controllers/NewsTableViewController+FetchNews.swift

```diff
@@ -0,0 +1,203 @@
+//
+//  NewsTableViewController+FetchNews.swift
+//  Canillitapp
+//
+//  Created by Ezequiel Becerra on 19/04/2019.
+//  Copyright © 2019 Ezequiel Becerra. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import ViewAnimator
+
+extension NewsTableViewController {
+
+    enum FetchMode {
+        case reload
+        case nextPage
+
+        func shouldAnimatePullToRefresh() -> Bool {
+            switch self {
+            case .reload:
+                return true
+            default:
+                return false
+            }
+        }
+
+        func shouldAnimateCells() -> Bool {
+            switch self {
+            case .reload:
+                return true
+            default:
+                return false
+            }
+        }
+    }
+
+    func setupPullToRefreshControl() {
+        //  Setup refresh control
+        guard let tableView = tableView else {
+            return
+        }
+
+        let refreshCtrl = UIRefreshControl()
+        tableView.refreshControl = refreshCtrl
+
+        refreshCtrl.tintColor = UIColor(red: 0.99, green: 0.29, blue: 0.39, alpha: 1.00)
+        refreshCtrl.addTarget(self, action: #selector(fetchReload), for: .valueChanged)
+
+        //  Had to set content offset because of UIRefreshControl bug
+        //  http://stackoverflow.com/a/31224299/994129
+        tableView.contentOffset = CGPoint(x: 0, y: -refreshCtrl.frame.size.height)
+    }
+
+    @objc func fetchReload() {
+        fetch(mode: .reload)
+    }
+
+    func fetch(mode: FetchMode) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/43ca6f2a532ef65a5ab17ff9bcaf840ce520a55f/Headlines/src/Controllers/NewsTableViewController+FetchNews.swift#L59"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 48 lines<br />
<code>function_body_length</code> <code>NewsTableViewController+FetchNews.swift:59</code></td>
    </tr>
  </tbody>
</table>